provisioning/WFGP-292_doc_galleon-pack.adoc

```diff
@@ -0,0 +1,164 @@
+---
+categories:
+- wf-galleon
+- maven
+- user-experience
+stability-level: community
+issue: https://github.com/wildfly/wildfly-proposals/issues/738
+feature-team:
+ developer: jmesnil
+ sme:
+  - jfdenise
+  - ehsavoie
+ outside-perspective:
+  - yersan
+---
+= Generate documentation for Galleon feature packs
+:author:            Jeff Mesnil
+:email:             jmesnil@ibm.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+*_"Javadoc" for feature packs_*
+
+The goal of that proposal is to provide documentation of the capabilities and resources that are provided by WildFly and its ecosystem of Galleon feature packs.
+This documentation is self-contained and automatically generated when the feature packs are built.
+
+== Overview
+
+At its core, WildFly is composed of extensions that provide management resources (grouped in subsystems and deployment resources).
+WildFly is providing a model reference for these resources at https://docs.wildfly.org/36/wildscribe/.
+
+Traditionally, this model reference was exhaustive and provided the full management capabilities of WildFly.
+It relied on its standalone-full-ha.xml configuration to provide it.
+
+However, over time, this approach has shown some cracks:
+
+* Some new subsystems are provided by other configurations (e.g. standalone-microprofile.xml). This was worked around by adding additional extensions when the model referenced is generated (in WildFly's `docs` module) but is out of sync.
+* WildFly is now composed of 2 Galleon packs (wildfly-ee-galleon-pack & wildfly-galleon-pack). Each Galleon pack provides different extensions and resources but this is not clear from the management model (e.g. which Galleon pack provides the `microprofile-config-smallrye` subsystem?)
+* Additional capabilities for WildFly are provided by other feature packs (mvc-krazo, grpc, wildfly-preview, etc.) that do not provide their model references and makes it more difficult for the users to understand which resources, attributes and operations are available to configure their installations. They have to rely on jboss-cli to get the information and while it is available, it does not make it very user-friendly.
```

Comment:
The mvc-krazo FP isn't a good example as the production code jar and galleon content that GitHub project provides are incorporated in the 'wildfly' feature pack. That project still builds its own FP but it's not really used.

BTW we do the same with vertx, so don't switch your example to that. ;)